js-envs/src/main/scala/org/scalajs/jsenv/nodejs/JSDOMNodeJSEnv.scala

```diff
@@ -65,46 +65,59 @@ class JSDOMNodeJSEnv private[jsenv] (
   protected trait AbstractDOMNodeRunner extends AbstractNodeRunner {
 
     protected def codeWithJSDOMContext(): Seq[VirtualJSFile] = {
-      val scriptsJSPaths = getLibJSFiles().map {
-        case file: FileVirtualFile => file.path
-        case file                  => libCache.materialize(file).getAbsolutePath
+      val scriptsFiles = (getLibJSFiles() :+ code).map {
+        case file: FileVirtualFile => file.file
+        case file                  => libCache.materialize(file)
       }
-      val scriptsStringPath = scriptsJSPaths.map('"' + escapeJS(_) + '"')
+      val scriptsURIsAsJSStrings = scriptsFiles.map { file =>
+        '"' + escapeJS(file.toURI.toASCIIString) + '"'
+      }
+      val scriptsURIsJSArray = scriptsURIsAsJSStrings.mkString("[", ", ", "]")
+
       val jsDOMCode = {
         s"""
            |(function () {
-           |  var jsdom;
-           |  try {
-           |    jsdom = require("jsdom/lib/old-api.js"); // jsdom >= 10.x
-           |  } catch (e) {
-           |    jsdom = require("jsdom"); // jsdom <= 9.x
-           |  }
+           |  var jsdom = require("jsdom");
```

Comment:
Can't we detect this somehow? It seems odd to break compatibility TBH.

Reply:
Sure we can. It's even easy: `typeof jsdom.JSDOM === "function"` will tell us whether it's jsdom >= 10. It's just that if we do that, we need to maintain two completely different implementations forever.

Every now and then, we require people to upgrade their version of sbt. I don't think it's unreasonable to demand that they upgrade to a newer version of another supporting infrastructure.

Reply:
Yes, you're right, the major version upgrade makes this case different than sbt. I'll adapt to keep support for v9.x.

Reply:
Done.

Locally tested with the same commands, with jsdom 9.12.0, 10.0.0 and 12.0.0.